### PR TITLE
Add dotty-doc to dist

### PR DESCRIPTION
**Pull request by felixmulder**

### CODE REVIEW

File: bin/dotd [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+CLASS_PATH="-classpath .$PSEP$DOTTY_LIB$PSEP$DOTTY_COMP$PSEP$DOTTY_DOC_LIB$PSEP$DOTTY_INTF"
+CLASS_PATH="$CLASS_PATH$PSEP$SCALA_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$FLEXMARK_LIBS"
+CLASS_PATH="$CLASS_PATH$PSEP$JACKSON_LIBS"
+CLASS_PATH="$CLASS_PATH$PSEP$LIQP_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$ANTLR_LIB$PSEP$ANTLR_RUNTIME_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$AUTOLINK_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$SNAKEYAML_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$ST4_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$JSOUP_LIB"
```
Comment (felixmulder) [inline bin/dotd]: Each line is an isolated dep, allowing us to delete them as they disappear over time.

Comment (DarkDimius) [inline bin/dotd]: Would be nice to test this script. Because it hardcodes list of dependencies it is very fragile

Comment (felixmulder) [inline bin/dotd]: Not sure any of the dist scripts are testable like this currently. We'd need to trigger `dist/pack-archive` then run the scripts I guess.

WDYT @liufengyun ?

Comment (liufengyun) [inline bin/dotd]: Currently we have a simple CI for check if a release can compile and run a hello world on OSX. The CI runs on Travis, and it's triggered everytime we update the formula.

We can also add `dotd` to the same CI to make sure it completes normally.

File: dist/bin/dotd [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+CLASS_PATH="$CLASS_PATH$PSEP$LIQP_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$ANTLR_LIB$PSEP$ANTLR_RUNTIME_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$AUTOLINK_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$SNAKEYAML_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$ST4_LIB"
+CLASS_PATH="$CLASS_PATH$PSEP$JSOUP_LIB"
+
+first_arg="$1"
+
+echo $JAVACMD dotty.tools.dottydoc.Main $CLASS_PATH
```
Comment (liufengyun) [inline dist/bin/dotd]: This line is just for testing?

File: dist/bin/dotd [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+eval exec "\"$JAVACMD\"" $CLASS_PATH dotty.tools.dottydoc.Main $@
```
Comment (liufengyun) [inline dist/bin/dotd]: BTW, do we need the Scala `classpath` for compiling the source? The `dotc` script has `Dscala.usejavacp=true` or `-Xbootclasspath/a:$classpath`, so it's not required.

File: dist/bin/dotd (hunk 2 above)
Comment (felixmulder) [inline dist/bin/dotd]: Ah yes, will remove

File: dist/bin/dotd (hunk 3 above)
Comment (liufengyun) [inline dist/bin/dotd]: @felixmulder I guess you missed this comment?

**Pull request merged**